### PR TITLE
backupccl: cleanup tests that relied on restoring unsupported backup fixtures

### PR DESCRIPTION
**Pull request by adityamaru**
As outlined in https://www.cockroachlabs.com/docs/v22.2/restoring-backups-across-versions.html,
we do not support restoring backups older than the minimum supported binary version of the cluster. For 23.1 this 
minimum supported binary version is going to be 22.2. This patch deletes or modifies existing tests that relied on backup
fixtures outside our minimum restoreable version. Some of the tests that exercised restores of
exceptional backups that could only be generated on pre-22.2 clusters have been deleted since users will have to
restore those backups into 22.2, and take another backup on 22.2 before restoring into 23.1. 

A follow up will enforce the minimum restoreable version during restore job execution.

Informs: https://github.com/cockroachdb/cockroach/issues/94295

### CODE REVIEW

**Review by msbutler — APPROVED** [review]
Valiant effort! Left a few nits that may increase test coverage.

File: pkg/ccl/backupccl/backup_test.go [hunk 1]
```diff
@@ -10876,3 +10876,104 @@ func TestExportResponseDataSizeZeroCPUPagination(t *testing.T) {
...
+		sqlDB.Exec(t, fmt.Sprintf(`RESTORE DATABASE test FROM LATEST IN $1 AS OF SYSTEM TIME %s`, ts[0]), localFoo)
+		// Just rendering the constraints loads and validates schema.
+		sqlDB.Exec(t, `SELECT * FROM pg_catalog.pg_constraint`)
+		sqlDB.Exec(t, `DROP DATABASE test`)
+
+		// Restore a couple tables, including parent but not child_pk.
+		sqlDB.Exec(t, `CREATE DATABASE test`)
+		sqlDB.Exec(t, fmt.Sprintf(`RESTORE test.circular FROM LATEST IN $1 AS OF SYSTEM TIME %s`, ts[0]), localFoo)
+		sqlDB.Exec(t, fmt.Sprintf(`RESTORE test.parent, test.child FROM LATEST IN $1 AS OF SYSTEM TIME %s  WITH skip_missing_foreign_keys`, ts[0]), localFoo)
+		sqlDB.Exec(t, `SELECT * FROM pg_catalog.pg_constraint`)
```
[the remark this answers is not in the document]
Comment (adityamaru) [reply pkg/ccl/backupccl/backup_test.go:10968]: Added checks below that a restore without skipping foreign keys fails for table child and child_pk.

File: pkg/ccl/backupccl/backup_test.go [hunk 2]
```diff
@@ -10876,3 +10876,104 @@ func TestExportResponseDataSizeZeroCPUPagination(t *testing.T) {
...
+	sqlDB.Exec(t, `RESTORE test.rev_times FROM LATEST IN $1 WITH into_db = 'ts'`, localFoo)
+	for _, ts := range sqlDB.QueryStr(t, `SELECT logical_time FROM ts.rev_times`) {
+		sqlDB.Exec(t, fmt.Sprintf(`RESTORE DATABASE test FROM LATEST IN $1 AS OF SYSTEM TIME %s`, ts[0]), localFoo)
+		// Just rendering the constraints loads and validates schema.
+		sqlDB.Exec(t, `SELECT * FROM pg_catalog.pg_constraint`)
+		sqlDB.Exec(t, `DROP DATABASE test`)
+
+		// Restore a couple tables, including parent but not child_pk.
+		sqlDB.Exec(t, `CREATE DATABASE test`)
+		sqlDB.Exec(t, fmt.Sprintf(`RESTORE test.circular FROM LATEST IN $1 AS OF SYSTEM TIME %s`, ts[0]), localFoo)
```
[the remark this answers is not in the document]
Comment (adityamaru) [reply pkg/ccl/backupccl/backup_test.go:10966]: Added a call to `SHOW CREATE` to ensure its the constraint we expect.

File: pkg/ccl/backupccl/testdata/backup-restore/regression-tests [hunk 3]
```diff
@@ -34,3 +34,51 @@ SELECT count(*) FROM [SHOW DATABASES] WHERE database_name = 'db1'
...
+subtest privileges-zone-config
+
+exec-sql
+CREATE DATABASE test;
+SET database = test;
+CREATE USER testuser;
+CREATE TABLE test_table();
+CREATE TABLE test_table2();
+GRANT ZONECONFIG ON DATABASE test TO testuser;
+GRANT ZONECONFIG ON test_table TO testuser;
```
[the remark this answers is not in the document]
Comment (adityamaru) [reply pkg/ccl/backupccl/testdata/backup-restore/regression-tests:47]: Hmm I think we stopped doing privileges inheritance. So a db privilege does not imply a table privilege afaik.

**Comment by adityamaru** [comment]
TFTR!

bors r=msbutler

**Comment by matthewtodd** [comment]
👋🏻 Hi, friends. In running the `v23.1.0-alpha.7` release just now, I ran into failures like the following in the [Windows bincheck](https://github.com/cockroachdb/cockroach/actions/runs/4472689494) script:

```
unable to create file pkg/ccl/backupccl/testdata/restore_mid_schema_change/exports/after/22.2/midaddcol/2023/03/10-143353.06/progress/BACKUP-CHECKPOINT-13cdcfcdccd2cfccd2cecfdfcecbc5ccccc5caccd1cdc9cbc8cbccdfd4cfcfcfcfdfaaabbcdf92c2d4c9ced1cac7cdcfc6cacccccbfffe: Filename too long
```

That is, the filenames introduced in this PR are long enough to break `git clone` on Windows. IIUC, there's nothing release blocking here, but if it's easy, it may be worth tightening these up. (A quick Google search shows the Windows filename limit to be ~256 characters.)

More officially, dev inf is aware and may look to introduce some git tooling to warn about long file names at some point in the future.

**Pull request merged**